packages/blockchain-wallet-v4-frontend/src/modals/SfoxExchangeData/Upload/index.js

```diff
@@ -78,7 +79,7 @@ class UploadContainer extends Component {
         submitForUpload={this.submitForUpload}
         handleStartClick={this.handleStartClick}
       />,
-      Failure: (msg) => <div>{msg.error}</div>,
+      Failure: (msg) => <Failure error={msg} />,
```

Comment:
How about adding componentDidCatch in addition to Failure cata? It would've saved us with todays' issue where cata would not.

Reply:
How would a componentDidCatch help us here? We have an ErrorBoundary component that wraps the entire app. In a componentDidCatch method wouldn't we wind up rendering the Failure template anyway?

Reply:
This would help if it won't be a failure but some code bug in the component like the last time.
It gives us an option to intercept it at form/module level and render error/retry message that does not kill the whole app.